zuul_alpha/core.py

```diff
@@ -341,13 +405,15 @@ def generate_key_pair(self):
             f.write(b64encode(ciphertext_blob))
 
         with open(self.public_key_file, 'w') as f:
-            f.write(public_key)
+            f.write(public_key_pem)
 
     def encrypt(self, secret_name, secret, public_key=None):
         '''Single secret encryptor
 
         Takes in a single key-value pair secret and encrypts it with the RSA
-        public key.
+        public key, UNLESS, the secret is too large for RSA payload size (214
```

Comment:
Even though it's a linear increase, it would double the amount of decrypt (compute) operations with no increased security. 

The pro would be for maintenance, that there would be parity in our encrypt operation.